core/src/types/vote_extensions/bridge_pool_roots.rs

```diff
@@ -0,0 +1,55 @@
+//! Vote extension types for adding a signature
+//! of the bridge pool merkle root to be added
+//! to storage. This will be used to generate
+//! bridge pool inclusion proofs for Ethereum.
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+
+use crate::proto::Signed;
+use crate::types::address::Address;
+use crate::types::key::common;
+use crate::types::key::common::Signature;
+use crate::types::storage::BlockHeight;
+
+/// A vote extension containing a validator's signature
+/// of the current root and nonce of the
+/// Ethereum bridge pool.
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+)]
+pub struct BridgePoolRootVext {
+    /// The validator sending the vote extension
+    pub validator_addr: Address,
```

Comment:
Should we have a TODO here linking to https://github.com/anoma/namada/issues/635 re. whether we need `validator_addr` field or not?

For initial release / Tendermint v0.37, I think we do definitely need it. If we were using real vote extensions, then we could work out which validator submitted a vote extension by looking at the Tendermint address of the validator which submitted the vote extension, and mapping that to their Namada address. We don't have that information to hand I guess while vote extensions are being packaged inside protocol transactions.